src/pop/blockchain/alt_block_tree.cpp

```diff
@@ -482,13 +500,21 @@ bool AltBlockTree::BlockPayloadMutator::isStatefulDuplicate(
   return false;
 }
 
+template <>
+bool AltBlockTree::BlockPayloadMutator::isStatefulDuplicate<VbkBlock>(
+    const VbkBlock::id_t& vbk_id) {
+  // make sure that provided VbkBlock does not contains in the existing
+  // VbkBlockTree state
+  return this->tree_.vbk().getBlockIndex(vbk_id) != nullptr;
```

Comment:
When we are performing stateful duplicate checks for alt block A payloads, the VBK tree view reflects block B, which isn't necessarily a direct ancestor, or an ancestor at all.